sys/cheri/c18n.h

```diff
@@ -85,6 +85,19 @@ struct cheri_c18n_info {
 	void * __kerncap	comparts;
 };
 
+/*
+ * The interface provided by the kernel via sysctl for compartmentalization
+ * monitoring tools such as procstat.
+ */
+#define	CHERI_C18N_COMPART_MAXNAME	56
+#define	CHERI_C18N_COMPART_LAST		-1
+
+struct cheri_c18n_compart {
```

Comment:
Hmm, I would model this on other kinfo_* structures.   I would embed a size field at the start and move the name field to the end.  The name could be packed in the exported form similar to what we do for `kinfo_file` and `kinfo_vmobject` sysctls now.  Userspace would unpack it back out into a simple array.